.github/workflows/build-and-test-multiplatform.yml

```
@@ -0,0 +1,796 @@
+    - name: Install Xvfb and Mesa
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y xvfb mesa-utils libgl1-mesa-dri
+
+    # Step: Start Xvfb
+    - name: Start Xvfb
+      run: Xvfb :99 -screen 0 1920x1080x24 &
+
+    # Step: Set DISPLAY environment variable
+    - name: Set DISPLAY environment variable
+      run: echo "DISPLAY=:99" >> $GITHUB_ENV
```

actionlint

shellcheck reported issue in this script: SC2086:info:1:23: Double quote to prevent globbing and word splitting

(shellcheck)